src/content/ui-components/components/color-highlight-button.mdx

```diff
@@ -0,0 +1,217 @@
+      disabled={!canColorHighlight}
+      aria-label={label}
+      aria-pressed={isActive}
+      style={{ backgroundColor: isActive ? highlightColor : 'transparent' }}
+    >
+      {label}
+      {shortcutKeys && <Badge>{parseShortcutKeys({ shortcutKeys })}</Badge>}
+    </button>
+  )
+}
+```
+
+#### Props
+
+| Name                  | Type                         | Default     | Description                                     |
+| --------------------- | ---------------------------- | ----------- | ----------------------------------------------- |
+| `editor`              | `Editor \| null`             | `undefined` | The Tiptap editor instance                      |
+| `highlightColor`      | `string`                     | `required`  | The highlight color to apply                    |
+| `label`               | `string`                     | `optional`  | Accessible label for the button                 |
+| `hideWhenUnavailable` | `boolean`                    | `false`     | Hides the button if highlight cannot be applied |
+| `onApplied`           | `({ color, label }) => void` | `undefined` | Callback fired after applying highlight         |
+
+#### Return Values
+
+| Name                    | Type            | Description                              |
+| ----------------------- | --------------- | ---------------------------------------- |
+| `isVisible`             | `boolean`       | Whether the button should be rendered    |
+| `isActive`              | `boolean`       | If the highlight is currently active     |
+| `canColorHighlight`     | `boolean`       | If the highlight can be applied          |
+| `handleColorHighlight`  | `() => boolean` | Function to apply the color highlight    |
+| `handleRemoveHighlight` | `() => boolean` | Function to remove the highlight         |
+| `label`                 | `string`        | Accessible label text for the button     |
+| `shortcutKeys`          | `string`        | Keyboard shortcut (Cmd/Ctrl + Shift + H) |
+| `Icon`                  | `React.FC`      | Icon component used (HighlighterIcon)    |
+
+## Utilities
+
+### `canColorHighlight(editor)`
+
+Checks if color highlight can be applied in the current editor state.
+
+```tsx
+import { canColorHighlight } from '@/registry/tiptap-ui/color-highlight-button'
+
+const canApply = canColorHighlight(editor)
+```
+
+### `isColorHighlightActive(editor, color?)`
+
+Checks if a color highlight is currently active in the selection.
+
+```tsx
+import { isColorHighlightActive } from '@/registry/tiptap-ui/color-highlight-button'
+
+const isActive = isColorHighlightActive(editor) // Any highlight active
+const isYellowActive = isColorHighlightActive(editor, 'var(--tt-color-highlight-yellow)') // Specific color active
+```
+
+### `pickHighlightColorsByValue(values: string[])`
+
+Filters predefined highlight colors (`HIGHLIGHT_COLORS`) by an array of color values. Useful for customizing which highlight options are available in your UI.
```

Comment:
I don't understand how to use this hook. How is it used with the highlight button? What should I then do with the `colors` variable?

It also does not explain how I access HIGHLIGHT_COLORS or how it is used.

I guess this is a low-level API?